web/src/settings_users.js

```diff
@@ -24,14 +24,35 @@ import * as settings_bots from "./settings_bots";
 import * as settings_config from "./settings_config";
 import * as settings_panel_menu from "./settings_panel_menu";
 import * as timerender from "./timerender";
+import * as ui_report from "./ui_report";
 import * as user_pill from "./user_pill";
+import * as user_profile from "./user_profile";
 
 const section = {
     active: {},
     deactivated: {},
     bots: {},
 };
 
+export function show_button_spinner($button) {
+    const $spinner = $button.find(".modal__spinner");
+    const dialog_submit_button_span_width = $button.find("span").width();
+    const dialog_submit_button_span_height = $button.find("span").height();
+    $button.prop("disabled", true);
+    $button.find("span").hide();
+    loading.make_indicator($spinner, {
+        width: dialog_submit_button_span_width,
+        height: dialog_submit_button_span_height,
+    });
+}
+
+export function hide_button_spinner($button) {
+    const $spinner = $button.find(".modal__spinner");
+    $button.prop("disabled", false);
+    $button.find("span").show();
+    loading.destroy_indicator($spinner);
+}
```

Comment:
These functions seem to be duplicates of `dialog_widget.ts` with a few of the class names changed. I think it probably makes sense to define reusable functions in `loading.ts` that both can call.